### ISSUE TITLE
Make it nicer to use rules from other npm packages

### ISSUE THREAD

**Issue by sindresorhus** [palantir/tslint#3436]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: CLI

#### Actual behavior

I want to use rules from an npm package, like [tslint-eslint-rules](https://github.com/buzinas/tslint-eslint-rules), in my shareable config, but I have to manually resolve the path to the rules directory to be able to use it:

```js
const path = require('path');

const getRuleDirectory = (name, directory) =>
	path.join(path.dirname(require.resolve(name)), directory || '');

module.exports = {
	rulesDirectory: [
		getRuleDirectory('tslint-eslint-rules', 'dist/rules'),
		getRuleDirectory('tslint-consistent-codestyle'),
		getRuleDirectory('tslint-microsoft-contrib')
	],
	…
};
```

*From https://github.com/sindresorhus/tslint-xo/blob/ecf13877ab4b4d37bc3e1616aacb027099382f47/index.js#L5-L13*

#### Expected behavior

Would be nice if I could just specify the package name, and TSLint would do the hard work of resolving the rules directory.

Something like this:

```js
module.exports = {
	rulesPackages: [
		'tslint-eslint-rules',
		'tslint-consistent-codestyle'
		'tslint-microsoft-contrib'
	],
	…
};
```

**Comment by ajafff** [comment]
Since TSLint 5.2.0 `rulesDirectory` uses node's module resolution.

The following should work:
```json
{
    "rulesDirectory": [
        "tslint-eslint-rules/dist/rules",
        "tslint-consistent-codestyle",
        "tslint-microsoft-contrib/dist/src"
    ]
}
```

The fact that you need to append the correct directory for `tslint-eslint-rules` and `tslint-microsoft-contrib` is caused by their package structure. If they let `"main"` point to a file in the rule directory, everything would work as expected. Though that will likely break users who rely on these paths.

**Comment by sindresorhus** [comment]
Oh, I missed that... I was looking at existing shareable configs:

- https://github.com/progre/tslint-config-airbnb/blob/a0f5a6b21f23e3488dfe2fef904aa1f135bc02f0/tslint.js#L5
- https://github.com/mrmlnc/tslint-config-xo/blob/144115976176cd7cd070e2f7621cf474f7508a01/index.js#L9

**Comment by mrmlnc** [comment]
Wow! Thanks. Will be fixed in my package :)

**Comment by sindresorhus** [comment]
@ajafff I tried using your solution, but it doesn't seem to be working. See:

- https://github.com/sindresorhus/tslint-xo/pull/3
- https://travis-ci.org/sindresorhus/tslint-xo/jobs/296670827

I'm just getting:

> Failed to load /home/travis/build/sindresorhus/tslint-xo/index.js: Could not find custom rule directory: tslint-eslint-rules/dist/rules

The plugins are a dependency of my shareable config, not the project I'm using the shareable config on? Is that a problem?

**Comment by ajafff** [comment]
That's kind of expected. Sorry I forgot to mention that earlier.
The rulesDirectory must contain an `index.js` or `index.json` file. It can be empty, it just needs to be there. That's how node resolves dependencies given a directory.

`tslint-consistent-codestyle` just works.

For `tslint-eslint-rules` you can simply put it in `"extends"`, because their preset does not contain any rules, just the rulesDirectory.
This does not apply to `tslint-microsoft-contrib` as they configure rules in their config preset. You probably want to open an issue there and link to this discussion.

**Comment by ajafff** [comment]
Related: https://github.com/buzinas/tslint-eslint-rules/issues/214

**Comment by JoshuaKGoldberg** [comment]
Is there anything actionable on TSLint's side here @ajafff / @sindresorhus? Or can this issue be closed here now that `extends` is allowed?

**Comment by aleclarson** [comment]
Doesn't node resolution also look for `package.json` and use the `main` field to validate the entry module's existence?
